components/sidebar_right/index.js

```diff
@@ -2,17 +2,66 @@
 // See License.txt for license information.
 
 import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+
+import PostStore from 'stores/post_store';
+
+import {showMentions, showPinnedPosts, showFlaggedPosts, closeRightHandSide} from 'actions/views/rhs';
+
+import {getRhsState, getSelectedPostId} from 'selectors/rhs';
+
+import {RHSStates} from 'utils/constants.jsx';
 
 import SidebarRight from './sidebar_right.jsx';
 
-function mapStateToProps(state, ownProps) {
+function mapStateToProps(state) {
+    const rhsState = getRhsState(state);
+
+    const channelId = state.views.rhs.channelId;
```

Comment:
With the recent change, this might be `selectedChannelId`

Reply:
thanks, using the selector now